_posts/2021-01-13-hapi-fhir.md

```diff
@@ -0,0 +1,454 @@
+
+    public  PatientProvidor() {
+    
+    }   
+    
+    @Override
+    public Class<? extends IBaseResource> getResourceType() {
+        return  Patient.class;
+    }
+
+    @Search
+    public List<Resource> searchPatient(HttpServletRequest request,
+                                        @OptionalParam(name= Patient.SP_BIRTHDATE) DateRangeParam birthDate,
+                                        @OptionalParam(name = Patient.SP_FAMILY) StringParam familyName,
+                                        @OptionalParam(name= Patient.SP_GENDER) StringParam gender,   
+                                        @OptionalParam(name= Patient.SP_GIVEN) StringParam givenName,
+                                        @OptionalParam(name = Patient.SP_IDENTIFIER) TokenParam identifier,
+                                        @OptionalParam(name= Patient.SP_NAME) StringParam name,
+                                        @OptionalParam(name = Patient.SP_RES_ID) TokenParam resid) {
+
+        List<Resource> results = patientDao.search(ctx, birthDate, familyName, gender, givenName, identifier, name);
+        return results;
+    }
+
+    @Read()
+    public Patient read(@IdParam IdType theId) {
+
+        Patient patient = patientDao.read(ctx,theId);
+        return patient;
+    }
+
+    @Create()
+    public MethodOutcome createPatient(HttpServletRequest theRequest, @ResourceParam Patient patient) {
+
+        MethodOutcome  method = new  MethodOutcome();
+        method.setCreated(true);
+        OperationOutcome  opOutcome = new  OperationOutcome();
+        method.setOperationOutcome(opOutcome)
+       
+        Patient  mongoPatient = patientDao.create(ctx, patient);
+        method.setId(mongoPatient.getIdElement());
+        method.setResource(mongoPatient);
+       
+        return method;
+    }
+
+    @Search()
+    public List<Resource> getAllPatients() {
+        List<Resource> results = patientDao.search(ctx);
+        return results;
+    }
+
+  
+
+    @Delete()
+    public void delete(@IdParam  IdType  theId) {
+        
+        patientDao.delete(ctx,theId);
+    }
+}
+
+```
+
+Now we configure our HAPI Server to support the patient resource.
+
+FhirRestfullServer.java
+
+```java
+
+@WebServlet("/*")
+public class FhirRestfullServer extends RestfulServer {
+    
+    private ApplicationContext applicationContext;
+
+    FhirRestfullServer(ApplicationContext context) {
+        this.applicationContext = context;
+    }
+
+    @Override
+    protected void initialize() throws ServletException {
+
+        super.initialize();
+        setFhirContext(FhirContext.forR4());
+        setResourceProviders(Arrays.asList(
+            applicationContext.getBean(PatientProvidor.class)));
+    }
+}
+
+```
+  
+In this section we have shown you how to implement a Patient resource on a HAPI FHIR server. 
+
+### Security
+
+Security is a very important part of setting up as a server especially when it comes to sensitive data like health information. 
```

Comment:
I think "as" is not neccesary here: "Security is a very important part of setting up as a server ..." --> "Security is a crucial part of setting up a server ..."